### DISCUSSION TITLE
aggregation with first instant alert and custom output

### DISCUSSION THREAD

**Discussion by paul-mesnilgrente** [jertel/elastalert2#426] [Q&A] [answered]
Hi, I'd like to create an alert which can group many matched documents.

I first thought that I could create loops with the `alert_text_type: alert_text_jinja` but the `type: frequency` contains only 1 matched document. Am I right?

Then I thought that `aggregation` was the solution but I'd like to get the alert to trigger as soon as something happens and then alert every x minutes with `realert`. Also, I don't think the output is customizable, or is it? I just saw the `summary_table_fields`, but it's still a table which I'd like to avoid.

**Comment by paul-mesnilgrente** [comment]
Maybe my question isn't clear enough. Do you have any questions?

**Comment by paul-mesnilgrente** [comment, marked as answer]
Could I have an answer on
> the `type: frequency` contains only 1 matched document to display. Am I right?

? I'm still bugged by this

**Reply by jertel** [reply]
I haven't used frequency much so I can't tell you a yes or no answer. The [frequency documentation](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#frequency), specifically the last parameter `attach_related` suggests that you may be right, as it seems to provide an optional means to include the other events with the alert.

**Reply by paul-mesnilgrente** [reply]
Ah! I didn't find this option. I'll try to investigate.
